manual/statements/batch/README.md

```diff
@@ -1,5 +1,32 @@
 ## Batch statements
 
-*Coming soon... In the meantime, see the javadoc for [BatchStatement].*
+Use [BatchStatement] to execute a set of queries as an atomic operation (refer to
+[Batching inserts, updates and deletes][batch_dse] to understand how to use batching effectively):
 
-[BatchStatement]: http://docs.datastax.com/en/drivers/java/3.6/com/datastax/driver/core/BatchStatement.html
+```java
+PreparedStatement preparedInsertExpense =
+    session.prepare(
+        "INSERT INTO cyclist_expenses (cyclist_name, expense_id, amount, description, paid) "
+            + "VALUES (:name, :id, :amount, :description, :paid)");
+SimpleStatement simpleInsertBalance =
+    new SimpleStatement("INSERT INTO cyclist_expenses (cyclist_name, balance) VALUES (?, 0) IF NOT EXISTS",
+        "Vera ADRIAN");
+
+BatchStatement batch = new BatchStatement()
+    .add(simpleInsertBalance)
+    .add(preparedInsertExpense.bind("Vera ADRIAN", 1, 7.95f, "Breakfast", false));
+
+session.execute(batch);
+```
+
+As shown in the examples above, batches can contain any combination of simple statements and bound
+statements. A given batch can contain at most 65536 statements. Past this limit, addition methods
+throw an `IllegalStateException`.
```

Comment:
Is it worth mentioning that batches are also limited by `batch_size_fail_threshold_in_kb` in cassandra.yaml?

Reply:
:+1: Added a section for that